Application recording flow

ceoaliongroo opened the pull request:
WIP: https://github.com/Gizra/public_education/issues/4

Commit:
- Modified view login
- Add login view html add-marker view

Commit:
- Fixed position login html

Commit:
Modified add-marker view:
* Fix wrapper class name for view in state == 'credentials'
* Add view for share view

Modified add-marker controller
* Add helper to set vars in the event onRecordUploaded()
* Change next state on onRecorded to 'credentials'

Commit:
Add server vars to access services

Commit:
Implementing states and adapting the flow to the record flow

amitaibu commented on app/scripts/controllers/addMarker.js:
```diff
@@ -71,11 +73,18 @@ angular.module('publicEducationApp')
      */
     $scope.onRecorded = function() {
-      $scope.setState('completed');
+      ($scope.user === null) ? $scope.setState('credentials') : $scope.setState('upload');
```
You should probably wrap this logic with `User.gettingUser()`

Commit:
Mocking upload file to server:
- Update the controller event upload
- Add external form
- Add view on state = 'upload' for view addMarker
- Fixed user promise logic in the event onRecorded with User

Commit:
Create draft view issue #6 only element displayed (no design)

ceoaliongroo commented on app/scripts/controllers/addMarker.js:
```diff
@@ -70,18 +69,38 @@ angular.module('publicEducationApp')
+      User.getUser().then(function(data) {
+        if (data === null) {
+          $scope.setState('credentials')
+        } else {
```
line break.

ceoaliongroo commented on app/scripts/controllers/addMarker.js:
```diff
@@ -70,18 +69,38 @@ angular.module('publicEducationApp')
+    }
 
+    $scope.$watch('state', function(newVal, oldVal) {
+      console.log(oldVal, ', ', newVal);
```
Remove.

ceoaliongroo commented on app/scripts/controllers/addMarker.js:
```diff
@@ -70,18 +69,38 @@ angular.module('publicEducationApp')
 
+      }
+
+      //Upload marker
```
=> `// Upload marker.`

ceoaliongroo commented on app/views/add-marker.html:
```diff
@@ -63,3 +64,46 @@
+      Your command has been recorded.
+    </p>
+    <p>
+      now, go tall your friends!
```
tall => tell.

ceoaliongroo commented on app/views/play-mode.html:
```diff
@@ -0,0 +1,31 @@
+    </div>
+  </div>
+  <div>
+    <img src="http://placehold.it/71x59" alt=""/>
```
Show a zoomed, no draggable, map instead of the placeholder.

Commit:
onDrag Event
* Disable keyboard to drag

amitaibu commented:
Closing, as already added a "play-marker" page, based on this code.
